virttest/utils_libvirt/libvirt_disk.py

```diff
@@ -119,3 +127,289 @@ def create_custom_metadata_disk(disk_path, disk_format,
     return custom_disk
+
+
+def get_images_with_xattr(vm):
+    """
+    Get the image path(s) which still having xattr left of a vm
+
+    :param vm: The vm to be checked
+    :return: The image path list which having xattr left
+    """
+    disks = vm.get_disk_devices()
+    dirty_images = []
+    check_cmd = "getfattr -m trusted.libvirt.security -d %s"
+    for disk in disks:
+        disk_path = disks[disk]['source']
+        getfattr_result = process.run(check_cmd % disk_path, shell=True)
+        if "selinux" in getfattr_result.stdout_text:
+            dirty_images.append(disk_path)
+            logging.debug("Image '%s' having xattr left: %s",
+                          disk_path, getfattr_result.stdout)
+    return dirty_images
+
+
+def clean_images_with_xattr(dirty_images):
+    """
+    Clean the images' xattr
+
+    :param dirty_images: The image list to be cleaned
+    """
+    for image in dirty_images:
+        get_attr_cmd = "getfattr -m trusted.libvirt.security -d %s" % image
+        output = process.run(get_attr_cmd, shell=True).stdout_text
+        attr_list = [line.split("=")[0] for line in output.split("\n") if len(line.split("=")) == 2
+                     and "trusted.libvirt" in line.split("=")[0]]
+        for attr in attr_list:
+            clean_cmd = "setfattr -x $s" % attr
+            process.run(clean_cmd, shell=True)
+
+
+def get_first_disk_source(vm):
+    """
+    Get disk source of first device
+
+    :param vm: VM instance
+    :return: first disk of first device.
+    """
+    first_device = vm.get_first_disk_devices()
+    first_disk_src = first_device['source']
+    return first_disk_src
+
+
+def make_relative_path_backing_files(vm, pre_set_root_dir=None, origin_image=None, origin_image_format="qcow2",
+                                     created_image_format="qcow2", back_chain_lenghth=4):
+    """
+    Create backing chain files of relative path for one image
+
+    :param vm: VM instance
+    :param pre_set_root_dir: preset root dir
+    :param origin_image: origin image
+    :param origin_image_format: original image format
+    :param created_image_format: created image format
+    :param back_chain_lenghth: back chain length
+    :return: absolute path of top active file and back chain files list
+    """
+    if pre_set_root_dir is None:
+        first_disk_source = get_first_disk_source(vm)
+        root_dir = os.path.dirname(first_disk_source)
+        basename = os.path.basename(first_disk_source)
+    else:
+        root_dir = pre_set_root_dir
+        basename = origin_image
+    sub_folders = list(map(chr, range(97, 97 + back_chain_lenghth)))
+    # Make external relative path backing files.
+    backing_file_dict = collections.OrderedDict()
+    for index in range(len(sub_folders)):
+        key = sub_folders[index]
+        cmd = "mkdir -p %s" % os.path.join(root_dir, key)
+        process.run(cmd, shell=True, ignore_status=False)
```

Comment:
I think we can use python utils to implement the mkdir

Reply:
@kylazhang 
Yes, but we need ignore error when folder exists already, but python utils will throw OSError.

os.mkdir() method in Python is used to create a directory named path with the specified numeric mode. This method raise FileExistsError if the directory to be created already exists.

Reply:
@kylazhang
Yes, but sometimes we need ignore error when folder exists already, but python utils will throw OSError.

os.mkdir() method in Python is used to create a directory named path with the specified numeric mode. This method raise FileExistsError if the directory to be created already exists.

Reply:
@chunfuwen you can use os.makedirs(directory, exist_ok=True)

Reply:
@kylazhang 

not anticipated as you suggest when using  os.makedirs(directory, exist_ok=True)
```
>>> import os
>>> root_dir = '/root/workdir/tp-libvirt'
>>> syslink_folder_list = ['a', 'b', 'c']
>>> for index in range(len(syslink_folder_list)):
...     key = syslink_folder_list[index]
...     folder_path = os.path.join(root_dir, key)
...     os.makedirs(folder_path, True)
... 
Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
  File "/usr/lib64/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/root/workdir/tp-libvirt/a'
```